src/prefs/DevicePrefs.cpp

```diff
@@ -410,6 +412,11 @@ bool DevicePrefs::Commit()
       else
          AudioIORecordingSource.Reset();
       AudioIORecordChannels.Write(mChannels->GetSelection() + 1);
+
+      AudioIORecordingDevice.Commit();
+      AudioIORecordingSourceIndex.Commit();
+      AudioIORecordingSource.Commit();
+      AudioIORecordChannels.Commit();
```

Comment:
I am not willing to accept this very special case fix of what might be an error of commit d9a873a that introduced SettingsTransaction.  It was not intended that individual settings variables should require commits like this.

Do you instead have an explanation why the line `mTransaction->Commit();` in PrefsDialog.cpp fails to have the effect it should?

Can you investigate, verify that d9a873a is the first bad commit in 3.2 showing this bug?